### PR TITLE
feat: load `SessionLens` manually via user command

### PR DESCRIPTION
**Pull request by ayamir**
https://github.com/ayamir/nvimdots/issues/742

### CODE REVIEW

**Comment by ayamir** [comment]
A more reasonable way is judge if setuped when execute `SessionLenShow` but I have not idea to get the condition about it. Maybe better solution is provide similar command in upstream itself.

**Review by Jint-lzxy — DISMISSED** [review]
I am hesitant to merge this PR into `main` b/c we have already set `auto-session` to load-on-demand. Perhaps @Saafo can add this to his/her custom config 👍

> A more reasonable way is judge if setuped

We have [`package.loaded`](https://www.lua.org/manual/5.1/manual.html#pdf-package.loaded) :)

**Comment by Saafo** [comment]
> Perhaps @Saafo can add this to his/her custom config 👍

Both are fine to me!

**Comment by ayamir** [comment]
> We have [package.loaded](https://www.lua.org/manual/5.1/manual.html#pdf-package.loaded) :)

Seems I can't set it like below.
```lua
	vim.api.nvim_create_user_command("SessionLenShow", function()
		if not package.loaded("telescope") then
			require("auto-session").setup_session_lens()
		end
		require("auto-session.session-lens").search_session()
	end, { nargs = 0 })
```

**Comment by Jint-lzxy** [comment]
> Seems I can't set it like below because the setup status of `session_lens` have no direct relation of whether telescope loaded or not.

We could check `auto-session.session-lens` instead ;)

**Comment by ayamir** [comment]
I tend to merge it into main.

File: lua/modules/configs/editor/auto-session.lua [hunk 1]
```diff
@@ -7,5 +7,18 @@ return function()
 		auto_save_enabled = true,
 		auto_restore_enabled = true,
 		auto_session_suppress_dirs = nil,
+		session_lens = {
+			-- If load_on_setup is set to false, one needs to eventually call `require("auto-session").setup_session_lens()` if they want to use session-lens.
```
Comment (Jint-lzxy) [inline lua/modules/configs/editor/auto-session.lua:11]:
```suggestion
			-- If `load_on_setup` is set to false, please use `SessionLensToggle` to manually load this plugin.
```

File: lua/modules/configs/editor/auto-session.lua [hunk 2]
```diff
@@ -7,5 +7,18 @@ return function()
...
+			load_on_setup = false,
+			theme_conf = { border = true },
+			previewer = false,
+		},
 	})
+
+	vim.api.nvim_create_user_command("SessionLenShow", function()
```
Comment (Jint-lzxy) [inline lua/modules/configs/editor/auto-session.lua:18]:
```suggestion
	vim.api.nvim_create_user_command("SessionLensToggle", function()
```

**Review by Jint-lzxy — COMMENTED** [review]
Adjusted phrasing 👍

**Pull request merged**